docs/ref/checks.txt

```diff
@@ -239,6 +239,8 @@ Related Fields
 * **fields.E338**: The intermediary model ``<through model>`` has no field
   ``<field name>``.
 * **fields.E339**: ``<model>.<field name>`` is not a foreign key to ``<model>``.
+* **fields.E340**: The field's intermediary table name clashes with
```

Comment:
Not sure which format you settled on, but this is different from the actual error message. It might be helpful to include the table name in the message.